app/controllers/admin/sponsors_controller.rb

```diff
@@ -4,11 +4,20 @@ class SponsorsController < Admin::BaseController
     load_and_authorize_resource :sponsor, through: :conference
     before_action :sponsorship_level_required, only: [:index, :new]
 
+    helper_method :add_swags
+
     def index
       authorize! :index, Sponsor.new(conference_id: @conference.id)
     end
 
-    def edit; end
+    def show
+      @sponsor.swag_index = @sponsor.swags.length
```

Comment:
Have you checked if this works for sponsors with, and sponsors without, swag?

Reply:
Yes. Even if there are no swags, `@sponsor.swag` is a hash that exists in every sponsor so `@sponsor.swag_index = @sponsor.swags.length ` equals to 0.